packages/web-components/fast-foundation/src/number-field/number-field.ts

```diff
@@ -363,6 +363,10 @@ export class FASTNumberField extends FormAssociatedNumberField {
      * @internal
      */
     public handleKeyDown(e: KeyboardEvent): boolean {
+        
+        if (this.readOnly) {
+            return true;
+        }
```

Comment:
We should likely check for disabled as well here.

```suggestion
        if (this.disabled || this.readOnly) {
            return true;
        }
```